From eef9ac8868dd53e2f770dcd5c69f21d36285e2c4 Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Thu, 8 Nov 2018 15:07:56 +0200
Subject: [PATCH 5075/5242] MLK-20189-2: ASoC: dsp: Replace atomic_dec_return
 with atomic_dec

commit  3f25235525809d56eb244bc76832b25c09bd3633 from
https://source.codeaurora.org/external/imx/linux-imx.git

Return value is not used so better use atomic_dec.
This also silences coverity warning CID3344689.

Reviewed-by: S.j. Wang <shengjiu.wang@nxp.com>
Reviewed-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_dsp.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/fsl/fsl_dsp.c b/sound/soc/fsl/fsl_dsp.c
index 6e61d69..d7960b5 100644
--- a/sound/soc/fsl/fsl_dsp.c
+++ b/sound/soc/fsl/fsl_dsp.c
@@ -503,7 +503,7 @@ static void dsp_mmap_close(struct vm_area_struct *vma)
 	pr_debug("xf_mmap_close: vma = %p, b = %p", vma, client);
 
 	/* ...decrement number of mapping */
-	atomic_dec_return(&client->vm_use);
+	atomic_dec(&client->vm_use);
 }
 
 /* ...memory map operations */
-- 
1.7.9.5

